source/server/active_stream_listener_base.cc

```diff
@@ -40,7 +40,7 @@ void ActiveStreamListenerBase::newConnection(Network::ConnectionSocketPtr&& sock
     return;
   }
   stream_info->setFilterChainName(filter_chain->name());
-  auto transport_socket = filter_chain->transportSocketFactory().createTransportSocket(nullptr);
+  auto transport_socket = filter_chain->transportSocketFactory().createDownstreamTransportSocket();
```

Comment:
The intention of this PR is to split createTransportSocket() into client method and server method. But is it intentional that the transport socket option param is removed from the server transport socket creation interface? We don't have config knob for server transport socket options right now, but if we want to do mutual TLS and server transport socket will need to verify cert, would server transport socket also take some transport socket options?

Reply:
Socket options mostly serve as a way to create multiple upstream transport sockets for distinct groups of downstream requests, one per each variation. It's not about configuring the transport socket itself, that's done through the factory.

Reply:
TransportSocketOptions::verifySubjectAltNameListOverride() is used in cert verification: https://github.com/envoyproxy/envoy/blob/469f7bbd5951a29dfc24a3a05ca01459b835da2c/source/extensions/transport_sockets/tls/cert_validator/default_validator.cc#L215

Could this be used by the server to match SAN values in a client cert? Or allowing per-connection override is non-sense? @RyanTheOptimist

Reply:
I think the listener filter chain match is used to match SAN values. I can imagine a more flexible transport socket that can adapt during negotiation, but Envoy doesn't have that yet.